`windows-2022` support

:wave: hello!

It looks like GitHub is rolling out `windows-2022` as the new default `windows-latest` according to [this changelog post](https://github.blog/changelog/2021-11-16-github-actions-windows-server-2022-with-visual-studio-2022-is-now-generally-available-on-github-hosted-runners/). I've seen some builds on `windows-latest` start to fail recently, like [this one](https://github.com/the-mikedavis/khepri/runs/5185856755?check_suite_focus=true), on the `setup-beam` step because the `ImageOS` env var is set to `win22` which is not yet mapped.

<details><summary>The full log for that build</summary>

```
2022-02-14T15:05:08.1124964Z Requested labels: windows-latest
2022-02-14T15:05:08.1125019Z Job defined at: the-mikedavis/khepri/.github/workflows/test.yaml@refs/heads/md-preemptive-var-info-inference
2022-02-14T15:05:08.1125042Z Waiting for a runner to pick up this job...
2022-02-14T15:05:09.9297790Z Job is waiting for a hosted runner to come online.
2022-02-14T15:05:19.4776996Z Job is about to start running on the hosted runner: GitHub Actions 4 (hosted)
2022-02-14T15:05:24.6449087Z Current runner version: '2.287.1'
2022-02-14T15:05:24.6479215Z ##[group]Operating System
2022-02-14T15:05:24.6479751Z Microsoft Windows Server 2022
2022-02-14T15:05:24.6480108Z 10.0.20348
2022-02-14T15:05:24.6480368Z Datacenter
2022-02-14T15:05:24.6480728Z ##[endgroup]
2022-02-14T15:05:24.6481081Z ##[group]Virtual Environment
2022-02-14T15:05:24.6481425Z Environment: windows-2022
2022-02-14T15:05:24.6481782Z Version: 20220207.1
2022-02-14T15:05:24.6482297Z Included Software: https://github.com/actions/virtual-environments/blob/win22/20220207.1/images/win/Windows2022-Readme.md
2022-02-14T15:05:24.6483045Z Image Release: https://github.com/actions/virtual-environments/releases/tag/win22%2F20220207.1
2022-02-14T15:05:24.6483546Z ##[endgroup]
2022-02-14T15:05:24.6483877Z ##[group]Virtual Environment Provisioner
2022-02-14T15:05:24.6484271Z 1.0.0.0-main-20220202-1
2022-02-14T15:05:24.6484553Z ##[endgroup]
2022-02-14T15:05:24.6485631Z ##[group]GITHUB_TOKEN Permissions
2022-02-14T15:05:24.6486482Z Actions: write
2022-02-14T15:05:24.6486849Z Checks: write
2022-02-14T15:05:24.6487277Z Contents: write
2022-02-14T15:05:24.6487607Z Deployments: write
2022-02-14T15:05:24.6487982Z Discussions: write
2022-02-14T15:05:24.6488267Z Issues: write
2022-02-14T15:05:24.6488589Z Metadata: read
2022-02-14T15:05:24.6488881Z Packages: write
2022-02-14T15:05:24.6489278Z Pages: write
2022-02-14T15:05:24.6489623Z PullRequests: write
2022-02-14T15:05:24.6489954Z RepositoryProjects: write
2022-02-14T15:05:24.6490404Z SecurityEvents: write
2022-02-14T15:05:24.6490705Z Statuses: write
2022-02-14T15:05:24.6491053Z ##[endgroup]
2022-02-14T15:05:24.6495167Z Secret source: Actions
2022-02-14T15:05:24.6495768Z Prepare workflow directory
2022-02-14T15:05:24.7538967Z Prepare all required actions
2022-02-14T15:05:24.7785612Z Getting action download info
2022-02-14T15:05:24.9764433Z Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
2022-02-14T15:05:25.4549030Z Download action repository 'erlef/setup-beam@v1' (SHA:730ac65cba0700bad37c0744de4aa916c16bb728)
2022-02-14T15:05:25.6624011Z Download action repository 'actions/cache@v2' (SHA:937d24475381cd9c75ae6db12cb4e79714b926ed)
2022-02-14T15:05:26.2507392Z ##[group]Run actions/checkout@v2
2022-02-14T15:05:26.2507935Z with:
2022-02-14T15:05:26.2508278Z   repository: the-mikedavis/khepri
2022-02-14T15:05:26.2508999Z   token: ***
2022-02-14T15:05:26.2509268Z   ssh-strict: true
2022-02-14T15:05:26.2509601Z   persist-credentials: true
2022-02-14T15:05:26.2509901Z   clean: true
2022-02-14T15:05:26.2510180Z   fetch-depth: 1
2022-02-14T15:05:26.2510466Z   lfs: false
2022-02-14T15:05:26.2510724Z   submodules: false
2022-02-14T15:05:26.2510994Z env:
2022-02-14T15:05:26.2511261Z   LATEST_OTP_RELEASE: 24
2022-02-14T15:05:26.2511556Z ##[endgroup]
2022-02-14T15:05:29.3749557Z Syncing repository: the-mikedavis/khepri
2022-02-14T15:05:29.3992657Z ##[group]Getting Git version info
2022-02-14T15:05:29.3994505Z Working directory is 'D:\a\khepri\khepri'
2022-02-14T15:05:29.4364860Z [command]"C:\Program Files\Git\bin\git.exe" version
2022-02-14T15:05:29.4603047Z git version 2.35.1.windows.2
2022-02-14T15:05:29.4654921Z ##[endgroup]
2022-02-14T15:05:29.4661582Z Deleting the contents of 'D:\a\khepri\khepri'
2022-02-14T15:05:29.4665511Z ##[group]Initializing the repository
2022-02-14T15:05:29.4674774Z [command]"C:\Program Files\Git\bin\git.exe" init D:\a\khepri\khepri
2022-02-14T15:05:29.6121854Z Initialized empty Git repository in D:/a/khepri/khepri/.git/
2022-02-14T15:05:29.6168243Z [command]"C:\Program Files\Git\bin\git.exe" remote add origin https://github.com/the-mikedavis/khepri
2022-02-14T15:05:29.6634518Z ##[endgroup]
2022-02-14T15:05:29.6635632Z ##[group]Disabling automatic garbage collection
2022-02-14T15:05:29.6661499Z [command]"C:\Program Files\Git\bin\git.exe" config --local gc.auto 0
2022-02-14T15:05:29.7236463Z ##[endgroup]
2022-02-14T15:05:29.7292307Z ##[group]Setting up auth
2022-02-14T15:05:29.7294446Z [command]"C:\Program Files\Git\bin\git.exe" config --local --name-only --get-regexp core\.sshCommand
2022-02-14T15:05:29.7524453Z [command]"C:\Program Files\Git\bin\git.exe" submodule foreach --recursive "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2022-02-14T15:05:31.8566762Z [command]"C:\Program Files\Git\bin\git.exe" config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-02-14T15:05:31.8913338Z [command]"C:\Program Files\Git\bin\git.exe" submodule foreach --recursive "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2022-02-14T15:05:32.4831021Z [command]"C:\Program Files\Git\bin\git.exe" config --local http.https://github.com/.extraheader "AUTHORIZATION: basic ***"
2022-02-14T15:05:32.5100344Z ##[endgroup]
2022-02-14T15:05:32.5101260Z ##[group]Fetching the repository
2022-02-14T15:05:32.5116138Z [command]"C:\Program Files\Git\bin\git.exe" -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +2403ae64d2d45f4b000016ba2643763ec5d785ae:refs/remotes/origin/md-preemptive-var-info-inference
2022-02-14T15:05:33.9891547Z remote: Enumerating objects: 49, done.        
2022-02-14T15:05:33.9892194Z remote: Counting objects:   2% (1/49)        
2022-02-14T15:05:33.9893253Z remote: Counting objects:   4% (2/49)        
2022-02-14T15:05:33.9893556Z remote: Counting objects:   6% (3/49)        
2022-02-14T15:05:33.9893857Z remote: Counting objects:   8% (4/49)        
2022-02-14T15:05:33.9894140Z remote: Counting objects:  10% (5/49)        
2022-02-14T15:05:33.9894431Z remote: Counting objects:  12% (6/49)        
2022-02-14T15:05:33.9894721Z remote: Counting objects:  14% (7/49)        
2022-02-14T15:05:33.9894986Z remote: Counting objects:  16% (8/49)        
2022-02-14T15:05:33.9895307Z remote: Counting objects:  18% (9/49)        
2022-02-14T15:05:33.9895706Z remote: Counting objects:  20% (10/49)        
2022-02-14T15:05:33.9896088Z remote: Counting objects:  22% (11/49)        
2022-02-14T15:05:33.9897865Z remote: Counting objects:  24% (12/49)        
2022-02-14T15:05:33.9898453Z remote: Counting objects:  26% (13/49)        
2022-02-14T15:05:33.9898803Z remote: Counting objects:  28% (14/49)        
2022-02-14T15:05:33.9899078Z remote: Counting objects:  30% (15/49)        
2022-02-14T15:05:33.9899360Z remote: Counting objects:  32% (16/49)        
2022-02-14T15:05:33.9899846Z remote: Counting objects:  34% (17/49)        
2022-02-14T15:05:33.9900117Z remote: Counting objects:  36% (18/49)        
2022-02-14T15:05:33.9900393Z remote: Counting objects:  38% (19/49)        
2022-02-14T15:05:33.9900667Z remote: Counting objects:  40% (20/49)        
2022-02-14T15:05:33.9900956Z remote: Counting objects:  42% (21/49)        
2022-02-14T15:05:33.9901609Z remote: Counting objects:  44% (22/49)        
2022-02-14T15:05:33.9901885Z remote: Counting objects:  46% (23/49)        
2022-02-14T15:05:33.9902181Z remote: Counting objects:  48% (24/49)        
2022-02-14T15:05:33.9902487Z remote: Counting objects:  51% (25/49)        
2022-02-14T15:05:33.9902774Z remote: Counting objects:  53% (26/49)        
2022-02-14T15:05:33.9903040Z remote: Counting objects:  55% (27/49)        
2022-02-14T15:05:33.9903319Z remote: Counting objects:  57% (28/49)        
2022-02-14T15:05:33.9905497Z remote: Counting objects:  59% (29/49)        
2022-02-14T15:05:33.9905802Z remote: Counting objects:  61% (30/49)        
2022-02-14T15:05:33.9906257Z remote: Counting objects:  63% (31/49)        
2022-02-14T15:05:33.9906533Z remote: Counting objects:  65% (32/49)        
2022-02-14T15:05:33.9906820Z remote: Counting objects:  67% (33/49)        
2022-02-14T15:05:33.9907117Z remote: Counting objects:  69% (34/49)        
2022-02-14T15:05:33.9907388Z remote: Counting objects:  71% (35/49)        
2022-02-14T15:05:33.9907667Z remote: Counting objects:  73% (36/49)        
2022-02-14T15:05:33.9908713Z remote: Counting objects:  75% (37/49)        
2022-02-14T15:05:33.9909286Z remote: Counting objects:  77% (38/49)        
2022-02-14T15:05:33.9909912Z remote: Counting objects:  79% (39/49)        
2022-02-14T15:05:33.9910212Z remote: Counting objects:  81% (40/49)        
2022-02-14T15:05:33.9910661Z remote: Counting objects:  83% (41/49)        
2022-02-14T15:05:33.9910927Z remote: Counting objects:  85% (42/49)        
2022-02-14T15:05:33.9911211Z remote: Counting objects:  87% (43/49)        
2022-02-14T15:05:33.9911480Z remote: Counting objects:  89% (44/49)        
2022-02-14T15:05:33.9911766Z remote: Counting objects:  91% (45/49)        
2022-02-14T15:05:33.9912050Z remote: Counting objects:  93% (46/49)        
2022-02-14T15:05:33.9912441Z remote: Counting objects:  95% (47/49)        
2022-02-14T15:05:33.9912724Z remote: Counting objects:  97% (48/49)        
2022-02-14T15:05:33.9912987Z remote: Counting objects: 100% (49/49)        
2022-02-14T15:05:33.9913291Z remote: Counting objects: 100% (49/49), done.        
2022-02-14T15:05:33.9913615Z remote: Compressing objects:   2% (1/46)        
2022-02-14T15:05:33.9913929Z remote: Compressing objects:   4% (2/46)        
2022-02-14T15:05:33.9914234Z remote: Compressing objects:   6% (3/46)        
2022-02-14T15:05:33.9914523Z remote: Compressing objects:   8% (4/46)        
2022-02-14T15:05:33.9914822Z remote: Compressing objects:  10% (5/46)        
2022-02-14T15:05:33.9915114Z remote: Compressing objects:  13% (6/46)        
2022-02-14T15:05:33.9915417Z remote: Compressing objects:  15% (7/46)        
2022-02-14T15:05:33.9915722Z remote: Compressing objects:  17% (8/46)        
2022-02-14T15:05:33.9916004Z remote: Compressing objects:  19% (9/46)        
2022-02-14T15:05:33.9916324Z remote: Compressing objects:  21% (10/46)        
2022-02-14T15:05:33.9916615Z remote: Compressing objects:  23% (11/46)        
2022-02-14T15:05:33.9916918Z remote: Compressing objects:  26% (12/46)        
2022-02-14T15:05:33.9917226Z remote: Compressing objects:  28% (13/46)        
2022-02-14T15:05:33.9917513Z remote: Compressing objects:  30% (14/46)        
2022-02-14T15:05:33.9927355Z remote: Compressing objects:  32% (15/46)        
2022-02-14T15:05:33.9932651Z remote: Compressing objects:  34% (16/46)        
2022-02-14T15:05:33.9936525Z remote: Compressing objects:  36% (17/46)        
2022-02-14T15:05:33.9941968Z remote: Compressing objects:  39% (18/46)        
2022-02-14T15:05:33.9949903Z remote: Compressing objects:  41% (19/46)        
2022-02-14T15:05:33.9959628Z remote: Compressing objects:  43% (20/46)        
2022-02-14T15:05:33.9981704Z remote: Compressing objects:  45% (21/46)        
2022-02-14T15:05:33.9982023Z remote: Compressing objects:  47% (22/46)        
2022-02-14T15:05:33.9989395Z remote: Compressing objects:  50% (23/46)        
2022-02-14T15:05:33.9996530Z remote: Compressing objects:  52% (24/46)        
2022-02-14T15:05:33.9999144Z remote: Compressing objects:  54% (25/46)        
2022-02-14T15:05:34.0025521Z remote: Compressing objects:  56% (26/46)        
2022-02-14T15:05:34.0038511Z remote: Compressing objects:  58% (27/46)        
2022-02-14T15:05:34.0055567Z remote: Compressing objects:  60% (28/46)        
2022-02-14T15:05:34.0057541Z remote: Compressing objects:  63% (29/46)        
2022-02-14T15:05:34.0076904Z remote: Compressing objects:  65% (30/46)        
2022-02-14T15:05:34.0078229Z remote: Compressing objects:  67% (31/46)        
2022-02-14T15:05:34.0097845Z remote: Compressing objects:  69% (32/46)        
2022-02-14T15:05:34.0098306Z remote: Compressing objects:  71% (33/46)        
2022-02-14T15:05:34.0098684Z remote: Compressing objects:  73% (34/46)        
2022-02-14T15:05:34.0099041Z remote: Compressing objects:  76% (35/46)        
2022-02-14T15:05:34.0102737Z remote: Compressing objects:  78% (36/46)        
2022-02-14T15:05:34.0103062Z remote: Compressing objects:  80% (37/46)        
2022-02-14T15:05:34.0103352Z remote: Compressing objects:  82% (38/46)        
2022-02-14T15:05:34.0103971Z remote: Compressing objects:  84% (39/46)        
2022-02-14T15:05:34.0112004Z remote: Compressing objects:  86% (40/46)        
2022-02-14T15:05:34.0116668Z remote: Compressing objects:  89% (41/46)        
2022-02-14T15:05:34.0117364Z remote: Compressing objects:  91% (42/46)        
2022-02-14T15:05:34.0118100Z remote: Compressing objects:  93% (43/46)        
2022-02-14T15:05:34.0118480Z remote: Compressing objects:  95% (44/46)        
2022-02-14T15:05:34.0118867Z remote: Compressing objects:  97% (45/46)        
2022-02-14T15:05:34.0119220Z remote: Compressing objects: 100% (46/46)        
2022-02-14T15:05:34.0119605Z remote: Compressing objects: 100% (46/46), done.        
2022-02-14T15:05:34.0907976Z remote: Total 49 (delta 6), reused 5 (delta 0), pack-reused 0        
2022-02-14T15:05:34.1903773Z From https://github.com/the-mikedavis/khepri
2022-02-14T15:05:34.1904622Z  * [new ref]         2403ae64d2d45f4b000016ba2643763ec5d785ae -> origin/md-preemptive-var-info-inference
2022-02-14T15:05:34.2462018Z ##[endgroup]
2022-02-14T15:05:34.2464408Z ##[group]Determining the checkout info
2022-02-14T15:05:34.2470295Z ##[endgroup]
2022-02-14T15:05:34.2470835Z ##[group]Checking out the ref
2022-02-14T15:05:34.2480960Z [command]"C:\Program Files\Git\bin\git.exe" checkout --progress --force -B md-preemptive-var-info-inference refs/remotes/origin/md-preemptive-var-info-inference
2022-02-14T15:05:34.3334003Z branch 'md-preemptive-var-info-inference' set up to track 'origin/md-preemptive-var-info-inference'.
2022-02-14T15:05:34.3336359Z Switched to a new branch 'md-preemptive-var-info-inference'
2022-02-14T15:05:34.3366187Z ##[endgroup]
2022-02-14T15:05:34.3785799Z [command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
2022-02-14T15:05:34.4045051Z '2403ae64d2d45f4b000016ba2643763ec5d785ae'
2022-02-14T15:05:34.4485622Z ##[group]Run erlef/setup-beam@v1
2022-02-14T15:05:34.4485969Z with:
2022-02-14T15:05:34.4486163Z   otp-version: 24
2022-02-14T15:05:34.4486429Z   rebar3-version: 3.16.1
2022-02-14T15:05:34.4486660Z   install-hex: true
2022-02-14T15:05:34.4486898Z   install-rebar: true
2022-02-14T15:05:34.4487125Z   version-type: loose
2022-02-14T15:05:34.4487385Z   disable_problem_matchers: false
2022-02-14T15:05:34.4487598Z env:
2022-02-14T15:05:34.4487813Z   LATEST_OTP_RELEASE: 24
2022-02-14T15:05:34.4488043Z ##[endgroup]
2022-02-14T15:05:34.5445029Z ##[error]Tried to map a target OS from env. variable 'ImageOS' (got win22), but failed. If you're using a self-hosted runner, you should set 'env': 'ImageOS': ... to one of the following: ['ubuntu18', 'ubuntu20', 'win19']
2022-02-14T15:05:34.5748197Z Post job cleanup.
2022-02-14T15:05:36.9350557Z [command]"C:\Program Files\Git\bin\git.exe" version
2022-02-14T15:05:36.9595925Z git version 2.35.1.windows.2
2022-02-14T15:05:36.9662090Z [command]"C:\Program Files\Git\bin\git.exe" config --local --name-only --get-regexp core\.sshCommand
2022-02-14T15:05:36.9954046Z [command]"C:\Program Files\Git\bin\git.exe" submodule foreach --recursive "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2022-02-14T15:05:37.6290467Z [command]"C:\Program Files\Git\bin\git.exe" config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-02-14T15:05:37.6527605Z http.https://github.com/.extraheader
2022-02-14T15:05:37.6569561Z [command]"C:\Program Files\Git\bin\git.exe" config --local --unset-all http.https://github.com/.extraheader
2022-02-14T15:05:37.6845500Z [command]"C:\Program Files\Git\bin\git.exe" submodule foreach --recursive "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2022-02-14T15:05:38.4291335Z Cleaning up orphan processes
```

</details>

Maybe it's as easy as treating win22 the same as win19?

--- a/README.md
+++ b/README.md
@@ -52,6 +52,7 @@ and Erlang/OTP.
 | ubuntu-18.04     | 17 - 24    | ✅
 | ubuntu-20.04     | 20 - 24    | ✅
 | windows-2019     | 21* - 24   | ✅
+| windows-2022     | 21* - 24   | ✅
 
 **Note** *: prior to 23, Windows builds are only available for minor versions, e.g. 21.0, 21.3, 22.0, etc.
 
@@ -65,6 +66,7 @@ uses that to download assets:
 | ubuntu18 | ubuntu-18.04
 | ubuntu20 | ubuntu-20.04
 | win19    | windows-2019
+| win22    | windows-2022
 
 as per the following example:
 
@@ -137,7 +139,7 @@ on: push
 
 jobs:
   test:
-    runs-on: windows-2019
+    runs-on: windows-2022
     steps:
       - uses: actions/checkout@v2
       - uses: erlef/setup-beam@v1

--- a/src/setup-beam.js
+++ b/src/setup-beam.js
@@ -405,6 +405,7 @@ function getRunnerOSVersion() {
     ubuntu18: 'ubuntu-18.04',
     ubuntu20: 'ubuntu-20.04',
     win19: 'windows-2019',
+    win22: 'windows-2022',
   }
   const containerFromEnvImageOS = ImageOSToContainer[process.env.ImageOS]
 

